runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/Exceptions.kt

```diff
@@ -82,12 +81,6 @@ public open class ServiceErrorMetadata : ErrorMetadata() {
         public val RequestId: AttributeKey<String> = AttributeKey("RequestId")
     }
 
-    /**
-     * The name of the service that sent this error response
-     */
-    public val serviceName: String
-        get() = attributes.getOrNull(SdkClientOption.ServiceName) ?: ""
-
```

Comment:
**Question**: Why was this removed?

Reply:
Because I moved `SdkClientOption` with the rest of the client stuff out of core. I looked through codegen, runtime, and generated clients. We don't ever populate this. I think we ought to remove it given that all generated exceptions inherit from a service specific exception already. I can be talked into a different direction though. I have questions/feels in general about `SdkClientOption` right now.